### DISCUSSION TITLE
Override server packets?

### DISCUSSION THREAD

**Discussion by spannercodes** [Minestom/Minestom#1344] [Help] [answered]
Hi, first time developing for Minestom here

Is it possible to intercept and modify or drop the server packets just before they are sent out?

I've had an idea on reducing network usage: by using a kind of occlusion culling, I could drop player movement packets when they aren't visible by a client. Assuming half of the players are behind the view of any given player, this could reduce the packets sent by 50%. Probably. I need to develop it to test it.

As stated above, I'm trying to find a way to listen to outgoing player position packets and for each client, drop the ones that are not in their view.

I've looked at the JavaDoc for an event but I can only find `PlayerPacketEvent` which is for _serverbound_ packets.

**Comment by KrystilizeNevaDies** [comment]
It might be a better idea to manually handle the viewers in this case.
You can do this with Player#addViewer and Player#removeViewer

**Comment by GreatWyrm** [comment, marked as answer]
There is the [PlayerPacketOutEvent](https://javadoc.minestom.net/net/minestom/server/event/player/PlayerPacketOutEvent.html) for clientbound packets, but you could also look into Viewable API (which determines whether or not entities can see each other).
